code/game/objects/items/blueprints.dm

```diff
@@ -140,36 +143,40 @@
 			else
 				usr << "<span class='warning'>Error! Please notify administration.</span>"
 				return
-	var/list/turf/turfs = res
+	var/list/turfs = res
 	var/str = trim(stripped_input(usr,"New area name:", "Blueprint Editing", "", MAX_NAME_LEN))
 	if(!str || !length(str)) //cancel
 		return
 	if(length(str) > 50)
 		usr << "<span class='warning'>The given name is too long.  The area remains undefined.</span>"
 		return
-	var/area/A = new
-	A.name = str
-	//var/ma
-	//ma = A.master ? "[A.master]" : "(null)"
-	//world << "DEBUG: create_area: <br>A.name=[A.name]<br>A.tag=[A.tag]<br>A.master=[ma]"
-	A.power_equip = 0
-	A.power_light = 0
-	A.power_environ = 0
-	A.always_unpowered = 0
-	A.valid_territory = 0
-	move_turfs_to_area(turfs, A)
-	A.SetDynamicLighting()
-
-	A.addSorted()
+
+	var/area/A
+	for(var/key in turfs)
+		if(key == str)
+			A = turfs[key]
+		if(turfs[key])
+			turfs -= turfs[key]
+			turfs -= key
+	if(A)
+		A.contents += turfs
+		A.SetDynamicLighting()
+	else
+		A = new

```

Comment:
If you can point me to something that creates new areas besides areaeditor, I'll do it.